server/controllers/list.js

```diff
@@ -705,13 +705,62 @@ const updateItemDetails = (req, resp) => {
   );
 };
 
+const cloneItem = (req, resp) => {
+  const { itemId } = req.body;
+  const { id: listId } = req.params;
+  const { _id: userId, displayName: userName } = req.user;
+  let newItemId;
+
+  List.findOne({
+    _id: listId,
+    'items._id': itemId,
+    $or: [{ ownerIds: userId }, { memberIds: userId }]
+  })
+    .exec()
+    .then(list => {
+      if (!list) {
+        throw new BadRequestException('List data not found.');
+      }
+
+      const { description, link, name } = list.items.id(itemId);
+
+      const item = new Item({
+        authorId: userId,
+        authorName: userName,
+        description,
+        link,
+        name
+      });
+
+      newItemId = item._id;
+      list.items.push(item);
+
+      return list.save();
+    })
+    .then(list =>
+      resp.status(200).send({
+        message: 'Item successfully cloned.',
+        item: responseWithItem(list.items.id(newItemId))
```

Comment:
I'm wondering do we need to send saved item, or we can just copy existing one from redux state and send new itemID only??

Reply:
We need new itemId and 'created-at', isVoted and VotesCount. Of course, I could send only these two first fields and set isVoted and VotesCount myself in cloneItem action but I think that now is more uniformly...